### PR TITLE
Refactor Action Selection 

### PR DESCRIPTION
**Pull request by chhex**

### CODE REVIEW

File: apg-patch-service-core/src/main/java/com/apgsga/microservice/patch/core/impl/PatchActionExecutorImpl.java [hunk 1]
```diff
@@ -30,30 +26,18 @@ public PatchActionExecutorImpl(SimplePatchContainerBean patchContainer) {
...
-			String res = (String) executeToStateAction.invoke(instance,new Object[]{patchNumber,toStatus,parameters});
-		} catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException | ClassNotFoundException e) {
-			throw ExceptionFactory.createPatchServiceRuntimeException(
-					"PatchActionExecutorImpl.execute.exception",
-					new Object[] { e.getMessage(), patchNumber, toStatus }, e);
-		}
+		Map<String,String> parameters = Maps.newHashMap();
+		parameters.put("target",stageMapping.getTarget());
+		parameters.put("stage",stage.getName());
+		PipelineInputAction pia = new PipelineInputAction(patchContainer);
```
Comment (chhex) [inline apg-patch-service-core/src/main/java/com/apgsga/microservice/patch/core/impl/PatchActionExecutorImpl.java:39]: It is either the PipelineInputAction or EntwicklungInstallationsbereitAction depending on the toState.

Comment (chhex) [reply apg-patch-service-core/src/main/java/com/apgsga/microservice/patch/core/impl/PatchActionExecutorImpl.java:39]: So it needs here either a Map with a Strategie Patttern or a simple If.

**Review by chhex — COMMENTED** [review]
I created a Pull request  to comment

**Comment by apgsga-jhe** [comment]
@chhex : not finish yet, but basically the structure will be like : https://github.com/apgsga-it/piper/pull/113/commits/d7c975eeb39724006b69d4b4586f828269a2f904

What do you think?

**Comment by apgsga-jhe** [comment]
@chhex : I made changes based on our discussion, this should be fine. I can ask you to approve it since you submitted the pull request ;) .... but I'll merge it when you give me the green light.

**Comment by apgsga-jhe** [comment]
@chhex  : Maybe you can let me know as soon as you'll have merge https://github.com/apgsga-it/piper/pull/114 ? Then I can merge this one as well.

**Pull request merged**